enclave/src/lib.rs

```diff
@@ -210,27 +212,123 @@ fn update_rsa_key(path: &str) -> Result<(), sgx_status_t> {
                     not_after: (now + RSA_DURATION) as time_t,
                 });
     }
+    println!("[+] Rsa key updated successfully!");
 
     write_rsa_key(path)?;
+    Ok(())
+}
+
+fn init_rsa_key() -> Result<(), sgx_status_t> {
+    match read_rsa_key(RSA_KEY_PATH) {
+        Ok(_) => (),
+        Err(_) => {
+            println!("[+] Trying to generate a new RSA key.");
+            update_rsa_key(RSA_KEY_PATH)?;
+        }
+    };
 
-    println!("Rsa key updated successfully !");
+    println!("[+] init_rsa_key() success!");
     Ok(())
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn init_rsa_key() -> sgx_status_t {
-    let rsa_key_path = "RSA_KEY";
+fn generate_quote() -> Option<Vec<u8>>{
+    let mut rsa_key_data: RSAKeyData = (*RSA_KEY_DATA.lock().unwrap()).clone().unwrap();
 
-    match read_rsa_key(rsa_key_path) {
+    let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap().as_secs();
+    // 5 minutes before the deadline still needs to update the key
+    if (rsa_key_data.not_befer < now as i64) | (rsa_key_data.not_after - ADVANCE_REFRESH_TIME < now as i64) {
```

Comment:
This line maybe too long, you can run `Cargo fmt` in the project directory to format your code.